packages/upload-client/src/unixfs.js

```diff
@@ -34,8 +34,12 @@ export function createFileEncoderStream(blob) {
   const unixfsWriter = UnixFS.createWriter({ writable, settings })
   const fileBuilder = new UnixFSFileBuilder('', blob)
   void (async () => {
-    await fileBuilder.finalize(unixfsWriter)
-    await unixfsWriter.close()
+    try {
+      await fileBuilder.finalize(unixfsWriter)
+      await unixfsWriter.close()
```

Comment:
Calling close on a writable stream that has errored causes this to throw an uncaught exception.

Reply:
☝️ This should probably be a code comment, also I don't think just logging an error is much better, browsers will log such errors and in node you can decide what to do. If we just log error things will appear to work but then break don't they ?

We should probably propagate error into returned readable somehow although I have to say I'm not sure how can we do that.

Any idea in what circumstances writable stream would error, it's not even exposed outside so if it errors perhaps it's problem in the `UnixFS` writer that can be fixed.

Reply:
I have removed and refactored the test so we don't encounter it.

What was causing it was the tests that were triggering the `ShardingStream` to error because a block is too big and it caused the CAR to exceed the shard size.